azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -874,7 +889,12 @@ private void prepareJobProperties(final ExecutableNode node) throws IOException
     // 2. Parent Flow Properties
     final ExecutableFlowBase parentFlow = node.getParentFlow();
     if (parentFlow != null) {
-      final Props flowProps = Props.clone(parentFlow.getInputProps());
+      // flow level runtime props have been already applied on parent input props
+      Props flowProps = Props.clone(parentFlow.getInputProps());
```

Comment:
This was surprising to me. Yes, it is documented in https://azkaban.github.io/azkaban/docs/latest/#property-override.

But there's an aspect that is not specifically documented: in flow 1.0 it's possible to create a a flow with a sub-flow like this:

`something.job`:
```
type=flow
flow.name=plainjob
someprop=something
```

`plainjob.job`
```
type=command
command=echo "${someprop}"
```

Now, if override is set for `someprop`, it also overrides the existing value, even though that value is set in a .job file (not the job in question itself though).

IMHO this is indeed surprising – I would assume this kind of property to be also treated as "existing" and not to be overridden, but of course this is a matter of choice.

If I'm not mistaken this is less confusing in flow 2.0 because it doesn't have the concept of `.properties` at all, instead it has been merged into the parent `config`.

Reply:
sorry, I didn't quite follow.